nvflare/lighter/poc_commands.py

```diff
@@ -120,28 +152,70 @@ def is_poc_ready(poc_workspace: str):
 
 def validate_poc_workspace(poc_workspace: str):
     if not is_poc_ready(poc_workspace):
-        print(f"workspace {poc_workspace} is not ready, please use poc --prepare to prepare poc workspace")
-        sys.exit(2)
+        raise Exception(f"workspace {poc_workspace} is not ready, please use poc --prepare to prepare poc workspace")
+
 
+def validate_gpu_ids(gpu_ids: list, host_gpu_ids: list):
+    for gpu_id in gpu_ids:
+        if gpu_id not in host_gpu_ids:
+            raise ValueError(f"gpu_id provided is not available in the host machine, available GPUs are {host_gpu_ids}")
+
+
+def get_host_gpu_ids() -> List[int]:
+    process = subprocess.Popen(["nvidia-smi", "--list-gpus"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    result = process.communicate()
+    st = process.poll()
+    rc = process.returncode
+    if rc > 0:
+        raise Exception("Failed to get host gpu device Ids", result[0])
+    else:
+        # 'GPU 0: NVIDIA GeForce RTX 3090 (UUID: GPU-xxxx-xxxx-xxxx-xxx)\n'
+        if result[0].startswith("GPU"):
+            gpus = result[0].split("\n")
+            gpu_ids = [int(gpu.split(":")[0].split(" ")[1]) for gpu in gpus[:-1]]
+        else:
+            gpu_ids = []
+
+    return gpu_ids
+
+
+def get_gpu_ids(user_input_gpu_ids, host_gpu_ids) -> List[int]:
+    if type(user_input_gpu_ids) == int and user_input_gpu_ids < 0:
```

Comment:
suggest `if user_input_gpu_ids == -1`

since that is a special case, or can default this value to -1